lib/ssl/src/ssl_handshake.erl

```diff
@@ -1151,6 +1202,30 @@ maybe_add_tls13_extensions({3,4},
 maybe_add_tls13_extensions(_, HelloExtensions, _, _, _) ->
     HelloExtensions.
 
+add_ocsp_stapling_extensions(
+    _Version, #{ocsp_stapling        := OcspStapling,
+                ocsp_responder_certs := OcspResponderCerts},
+    OcspNonce, HelloExtensions) ->
+    case OcspStapling of
+        false ->
+            HelloExtensions;
+        true ->
+            OcspResponderList = get_ocsp_responder_list(OcspResponderCerts, []),
+            OcspRequestExtns = public_key:ocsp_extensions(OcspNonce),
+            Req = #ocsp_status_request{responder_id_list = OcspResponderList,
+                                       request_extensions = OcspRequestExtns},
+            CertStatusReqExtn = #certificate_status_request{
+                status_type = ?CERTIFICATE_STATUS_TYPE_OCSP,
+                request = Req
+            },
+            HelloExtensions#{ocsp_stapling => CertStatusReqExtn}
+    end.
+
+get_ocsp_responder_list([], Acc) ->
```

Comment:
This is a pattern that we usually use to hide the initialization of the list accumulator:
```erlang
get_ocsp_responder_list(L) ->
    get_ocsp_responder_list(L, []).
%%
get_ocsp_responder_list([], Acc) -> ...
```

Reply:
> usually

done